src/test/resources/specs/name-matching/analysis/Last_names_combination_meta_data.feature

```diff
@@ -0,0 +1,390 @@
+      | FIRST     | 0       | 6       | ENTIRE            |
+      | LAST      | 0       | 9       | ENTIRE            |
+
+
+  Scenario: Meta-Data for Apostrophed last name is tried without apostrophes 2
+    Given HMRC has the following individual records
+      | First name      | Last name       | Date of Birth | nino        |
+      | Arthur          | OBobbins        | 9999-12-31    | NR123456C   |
+    When an income request is made with the following identity
+      | First name    | Arthur     |
+      | Last name     | O'Bobbins  |
+      | Date of Birth | 9999-12-31 |
+      | nino          | NR123456C  |
+    Then meta-data was logged following a successful match
+    And the meta-data contains the following input name information
+      | nameType  | index | diacritics  | umlauts | abbreviation  | nameSplitter  | uniCodeBlocks | length  |
+      | FIRST     | 0     | false       | false   | false         | false         | BASIC_LATIN   | 6       |
+      | LAST      | 0     | false       | false   | false         | true          | BASIC_LATIN   | 9       |
+    And the meta-data indicates that the following generators were used
+      | SPLITTERS_REMOVED           |
+      | ENTIRE_NON_ALIAS_NAME       |
+    And the meta-data contains the following name derivation information
+      | section   | index   | length  | derivationActions |
+      | FIRST     | 0       | 6       | ENTIRE            |
+      | LAST      | 0       | 8       | ENTIRE            |
+
+
+  Scenario: Meta-Data for Apostrophed last name is tried without apostrophes 3
+    Given HMRC has the following individual records
+      | First name      | Last name       | Date of Birth | nino        |
+      | Arthur          | Bobbins         | 9999-12-31    | NR123456C   |
+    When an income request is made with the following identity
+      | First name    | Arthur     |
+      | Last name     | O'Bobbins  |
+      | Date of Birth | 9999-12-31 |
+      | nino          | NR123456C  |
+    Then meta-data was logged following a successful match
+    And the meta-data contains the following input name information
+      | nameType  | index | diacritics  | umlauts | abbreviation  | nameSplitter  | uniCodeBlocks | length  |
+      | FIRST     | 0     | false       | false   | false         | false         | BASIC_LATIN   | 6       |
+      | LAST      | 0     | false       | false   | false         | true          | BASIC_LATIN   | 9       |
+    And the meta-data indicates that the following generators were used
+      | SPLITTERS_REPLACED           |
+      | NAME_MATCHING                |
+    And the meta-data contains the following name derivation information
+      | section   | index   | length  | derivationActions |
+      | FIRST     | 0       | 6       | ORIGINAL          |
+      | LAST      | 0       | 7       | RIGHT_OF_SPLIT    |
+
+
+  Scenario: Meta-Data for Apostrophed last name is tried without apostrophes 4
```

Comment:
I know this is not related to the meta-data collection, but I wonder if the match for this test is correct - I don't think candidate name "Arthur O'Bobbins" would get a match with HMRC name "Arthur O".

On a plus note, the meta-data makes it much easier to understand edge cases like this :)

Reply:
I'll chase this up with Owain.